velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -279,24 +279,42 @@ std::shared_ptr<const ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
               maxDefine);
         }
         case thrift::ConvertedType::MAP_KEY_VALUE: {
-          // child of MAP
-          VELOX_CHECK_EQ(
-              schemaElement.repetition_type,
-              thrift::FieldRepetitionType::REPEATED);
-          assert(children.size() == 2);
-          auto childrenCopy = children;
-          return std::make_shared<const ParquetTypeWithId>(
-              TypeFactory<TypeKind::MAP>::create(
-                  children[0]->type(), children[1]->type()),
-              std::move(childrenCopy),
-              curSchemaIdx, // TODO: there are holes in the ids
-              maxSchemaElementIdx,
-              ParquetTypeWithId::kNonLeaf, // columnIdx,
-              std::move(name),
-              std::nullopt,
-              std::nullopt,
-              maxRepeat,
-              maxDefine);
+          if (children.size() == 1) {
+            // MAP_KEY_VALUE is incorrectly used in place of MAP
+            // Backward-compatibility rules described in spec
+            // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists
```

Comment:
Should the link be this? https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules-1

Reply:
Updated the link, thanks.